infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/utils/LogReplicationConfigManager.java

```diff
@@ -143,9 +143,18 @@ public LogReplicationConfig getUpdatedConfig() {
     }
 
     private boolean registryTableHasNewEntries() {
```

Comment:
can this method itself be made `synchronized` perhaps ?

Reply:
I think making this method `synchronized` might not resolve the issue, from what I see the issue comes from we are updating `lastRegistryTableLogTail` (and also, return the true/false) before actually processing the new registry table entries. Will think more about this issue in my next PR